pkg/controllers/termination/suite_test.go

```diff
@@ -89,18 +88,83 @@ var _ = Describe("Termination", func() {
 					v1alpha2.ProvisionerNamespaceLabelKey: "default",
 				},
 			})
-			ExpectCreated(env.Client, node)
-			ExpectCreated(env.Client, test.Pod(test.PodOptions{
+			podEvict := test.Pod(test.PodOptions{
+				NodeName: node.Name,
+			})
+			podSkip := test.Pod(test.PodOptions{
 				NodeName:    node.Name,
 				Tolerations: []v1.Toleration{{Key: v1.TaintNodeUnschedulable, Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoSchedule}},
-			}))
+			})
+			ExpectCreated(env.Client, node)
+			ExpectCreated(env.Client, podEvict, podSkip)
 
+			// Trigger Termination Controller
 			Expect(env.Client.Delete(ctx, node)).To(Succeed())
 			node = ExpectNodeExists(env.Client, node.Name)
 			ExpectReconcileSucceeded(controller, node)
-			pods := &v1.PodList{}
-			Expect(env.Client.List(ctx, pods, client.MatchingFields{"spec.nodeName": node.Name})).To(Succeed())
-			Expect(pods.Items).To(HaveLen(1))
+
+			// Expect podToEvict to be evicting, and delete it
+			podEvict = ExpectPodExists(env.Client, podEvict.Name, podEvict.Namespace)
+			Expect(podEvict.GetObjectMeta().GetDeletionTimestamp().IsZero()).To(Equal(false))
+			ExpectDeleted(env.Client, podEvict)
+			// Expect podToSkip to not be evicting
+			podSkip = ExpectPodExists(env.Client, podSkip.Name, podSkip.Namespace)
+			Expect(podSkip.GetObjectMeta().GetDeletionTimestamp().IsZero()).To(Equal(true))
+
+			// Reconcile to delete node
+			node = ExpectNodeExists(env.Client, node.Name)
+			ExpectReconcileSucceeded(controller, node)
+			ExpectNotFound(env.Client, node)
+		})
+		It("should not terminate nodes that have a do-not-evict pod", func() {
+			node := test.NodeWith(test.NodeOptions{
+				Finalizers: []string{v1alpha2.KarpenterFinalizer},
+				Labels: map[string]string{
+					v1alpha2.ProvisionerNameLabelKey:      "default",
+					v1alpha2.ProvisionerNamespaceLabelKey: "default",
+				},
+			})
+			pods := []*v1.Pod{
+				test.Pod(test.PodOptions{
```

Comment:
consider naming these pods similar to the previous test, I think it will clean things up.